mbed_host_tests/__init__.py

```diff
@@ -494,14 +479,25 @@ def execute(self):
                 self.notify("HOST: Error! No host test name defined in preamble")
                 self.print_result(self.RESULT_ERROR)
 
-            if result is not None:
-                self.print_result(result)
+            if self.result is not None:
+                self.print_result(self.result)
             else:
                 self.notify("HOST: Passive mode...")
         except Exception, e:
             print str(e)
             self.print_result(self.RESULT_ERROR)
 
+    def run_test(self, host_test_name):
+        try:
+            self.test_supervisor = get_host_test(host_test_name)
```

Comment:
Since ```if is_host_test(CONFIG["host_test_name"]):``` checks that host test exists before ```run_test ()``` calls ```get_host_test()``` it should always work. But at this line if feels unsafe to call ```test_supervisor.setup()``` without checking.
Perhaps we can call ```get_host_test()``` only and before calling ```run_test(test_supervisor)``` or put an assert here like
``` assert self.test_supervisor is not None, "get_host_test() return None"```